ui/app/lib/kv-object.js

```diff
@@ -1,13 +1,18 @@
 import Ember from 'ember';
 
+const { typeOf, guidFor } = Ember;
+
 export default Ember.ArrayProxy.extend({
   fromJSON(json) {
-    const contents = Object.keys(json || []).map(key => {
+    if (json && typeOf(json) !== 'object') {
+      throw new Error('Vault expects data to be formatted as an JSON object.');
```

Comment:
I was thinking this is already the advanced mode so there would be some idea of what they're doing. The index thing was happening because we were using `Object.keys` on an array, but now we short circuit that, so it would take quite a bit of fiddling to get back there. 

What sort of extra explanation were you thinking? Maybe: "The input is valid JSON, but Vault requires the top-level structure to be a JSON Object and currently it is not." Or something like that?